build(docs-infra): ignore `__`-prefixed and `ng*Def` members in docs

gkalpak opened the pull request:
☝️

gkalpak committed:
fix(docs-infra): ignore `ng*Def` members in API docs

`ng*Def` properties (such as `ngInjectorDef`) are not considered part of
the public API and should not appear in the API docs. This commit adds a
filter to remove these properties from the docs metadata.

mary-poppins commented:
You can preview d416be6 at https://pr31378-d416be6.ngbuilds.io/.

mary-poppins commented:
You can preview 1126e03 at https://pr31378-1126e03.ngbuilds.io/.

IgorMinar reviewed (approved):
looks good. thanks for all the tests.

the only nit: the commit message could be a bit more descriptive

mary-poppins commented:
You can preview 0958aeb at https://pr31378-0958aeb.ngbuilds.io/.

gkalpak committed:
fixup! fix(docs-infra): ignore `ng*Def` members in API docs

gkalpak committed:
docs(core): mark `EventEmitter#__isAsync` as internal to hide from API docs

The `__isAsync` property is not part of the public API and should not
appear in the API docs.

mary-poppins commented:
You can preview 2e463ea at https://pr31378-2e463ea.ngbuilds.io/.
